frontend/src/app/modules/workspace/settings/settings.component.ts

```diff
@@ -0,0 +1,220 @@
+    this.initValues();
+    this.initForms();
+  }
+
+  initValues() {
+    let token = localStorage.getItem('accessToken') || '';
+
+    if (token) {
+      let decoded: { name: string; email: string; id: string } =
+        jwt_decode(token);
+
+      this.name = decoded.name;
+      this.oldName = this.name;
+      this.oldEmail = decoded.email;
+      this.userId = decoded.id;
+    }
+  }
+
+  initForms() {
+    this.formAvatar = this.formBuilder.group({
+      workspaceName: [
+        this.name,
+        {
+          validators: [
+            Validators.required,
+            Validators.minLength(3),
+            Validators.maxLength(20),
+            Validators.pattern("^[a-zA-Z`'][a-zA-Z-`' ]+[a-zA-Z`']?$"),
+          ],
+        },
+      ],
+    });
+
+    this.formPassword = this.formBuilder.group({
+      email: [
+        this.oldEmail,
+        {
+          validators: [Validators.required, Validators.email],
+        },
+      ],
+      passOld: [
+        ,
+        {
+          validators: [
+            Validators.required,
+            Validators.minLength(8),
+            Validators.maxLength(20),
+            Validators.pattern('^(?=.*[0-9])(?=.*[a-zA-Z])([a-zA-Z0-9]+)$'),
+          ],
+        },
+      ],
+      passNew: [
+        ,
+        {
+          validators: [
+            Validators.required,
+            Validators.minLength(8),
+            Validators.maxLength(20),
+            Validators.pattern('^(?=.*[0-9])(?=.*[a-zA-Z])([a-zA-Z0-9]+)$'),
+          ],
+        },
+      ],
+    });
+  }
+
+  errorHandler(event: Event) {
+    const target = event.target as HTMLInputElement;
+    target.src = '../../assets/icons/test-user-logo.png';
+  }
+
+  saveNewInfo() {
+    let workspaceName = this.formAvatar.controls['workspaceName'].value;
+
+    const data = new FormData();
+    data.append('id', this.userId);
+    data.append('workspaceName', workspaceName);
+    data.append('avatar', this.imageFile);
+
+    this.userService.updateInfo('Update', data).subscribe({
+      next: () => {
+        window.alert('done');
+      },
+      error: () => {
+        window.alert('error');
+      },
+    });
+  }
+
+  onSelect(event: { addedFiles: File[] }) {
+    let added = event.addedFiles;
+    if (added?.length > 0) {
+      this.imageFile = added[0];
+
+      const reader = new FileReader();
+      reader.addEventListener(
+        'load',
+        () => (this.avatar = reader.result as string)
+      );
+      reader.readAsDataURL(this.imageFile);
+    }
+  }
+
+  saveNewInfoCancel() {
+    this.formAvatar.controls['workspaceName'].setValue(this.oldName);
+    this.avatar = '';
+  }
+
+  public handleFileInput(event: Event) {
+    let target = event.target as HTMLInputElement;
+    this.imageFile = target.files?.item(0) as File;
+
+    if (!this.imageFile) {
+      target.value = '';
+      return;
+    }
+
+    if (this.imageFile.size / 1024 / 1024 > 5) {
+      window.alert('Image can`t be heavier than ~5MB');
```

Comment:
We have angular notifications, can you please use them here
It all places

Reply:
Notifications in the dev branch, after approval, will merge and add